plans/action/install.pp

```diff
@@ -203,27 +208,32 @@
   # Create csr_attributes.yaml files for the nodes that need them. Ensure that
   # if a csr_attributes.yaml file is already present, the values we need are
   # merged with the existing values.
-
-  run_plan('peadm::util::insert_csr_extension_requests', $master_target,
-    extension_requests => {
-      peadm::oid('peadm_role')               => 'puppet/master',
-      peadm::oid('peadm_availability_group') => 'A',
-    },
-  )
-
-  run_plan('peadm::util::insert_csr_extension_requests', $puppetdb_database_target,
-    extension_requests => {
-      peadm::oid('peadm_role')               => 'puppet/puppetdb-database',
-      peadm::oid('peadm_availability_group') => 'A',
-    },
-  )
-
-  run_plan('peadm::util::insert_csr_extension_requests', $puppetdb_database_replica_target,
-    extension_requests => {
-      peadm::oid('peadm_role')               => 'puppet/puppetdb-database',
-      peadm::oid('peadm_availability_group') => 'B',
-    },
-  )
+  parallelize([$master_target, $puppetdb_database_target, $puppetdb_database_replica_target]) |$target| {
```

Comment:
I made a comment below, but thinking again: we should probably `.peadm::flatten_compact()` this anyway, yes? Otherwise it is an array of arrays, since every `$*_target` variable here is an array. Is that behavior safe/documented? Even if so, I like the idea of cleaning up the input to be a single array of Target objects.

Reply:
I agree. Will add the function to `parallelize()` calls when I run through to test other architectures

Reply:
Noticed that `$pe_installer_targets` is made of the same elements as array, will just use that instead, is already using `peadm::flatten_compact()`